File: from-3001-to-3100/3042-count-prefix-and-suffix-pairs-i/solution.go
func countPrefixSuffixPairs(words []string) int {
    isPrefixAndSuffix := func(s1, s2 string)bool {
        n1, n2 := len(s1), len(s2)
        if n1 <= n2 && s1 == s2[:n1] && s1 == s2[n2-n1:] {
            return true
        }
        return false
    }
    result := 0
    for i := 0; i < len(words) - 1; i ++ {
        for j := i + 1; j < len(words); j ++ {
            if isPrefixAndSuffix(words[i], words[j]) {
                result += 1
            }
        }
    }
    return result
}

File: from-3001-to-3100/3042-count-prefix-and-suffix-pairs-i/solution.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        def is_prefix_and_suffix(s1: str, s2: str) -> bool:
            n1, n2 = len(s1), len(s2)
            if n1 <= n2 and s1 == s2[:n1] and s1 == s2[n2 - n1:]:
                return True
            return False
        
        result = 0
        for i, word1 in enumerate(words):
            for word2 in words[i + 1:]:
                if is_prefix_and_suffix(word1, word2):
                    result += 1
        return result